samples/payload/payload.cpp

```diff
@@ -0,0 +1,66 @@
+#include <string>
+#include <windows.h>
+#include <detours.h>
+#include <winrt/base.h>
+
+#include "payloadguid.hpp"
+
+std::wstring GetProcessFileName(HANDLE process)
+{
+	DWORD exeLocation_size = MAX_PATH + 1;
+
+	std::wstring exeLocation;
+	exeLocation.resize(exeLocation_size);
+
+	winrt::check_bool(QueryFullProcessImageNameW(process, 0, exeLocation.data(), &exeLocation_size));
+
+	exeLocation.resize(exeLocation_size);
+	return exeLocation;
+}
+
+template<typename T>
+volatile T* InjectPayload(HANDLE hProcess, T payload, REFGUID guid)
+{
+	auto newPayload = static_cast<volatile T*>(DetourCopyPayloadToProcessEx(hProcess, guid, &payload, sizeof(payload)));
+	winrt::check_bool(newPayload);
+
+	return newPayload;
+}
+
+int main()
+{
+	std::wstring target = GetProcessFileName(GetCurrentProcess());
+	target.erase(target.rfind(L'\\') + 1);
+	target += L"payloadtarget.exe";
+
+	STARTUPINFOW si = { sizeof(si) };
+	PROCESS_INFORMATION pi;
+	winrt::check_bool(CreateProcessW(target.c_str(), nullptr, nullptr, nullptr, false, CREATE_SUSPENDED, nullptr, nullptr, &si, &pi));
```

Comment:
winrt::check_bool throws, seems like we are missing a try/catch block to report failure?

Reply:
If an uncaught exception is thrown, the process will automatically exit with an non-0 code. Since cppwinrt needs to go away (because it requires C++17) I'll look into adding some logging.